TileMap.gd

```diff
@@ -5,34 +5,50 @@ var overworld
 var tile_size = get_cell_size()
 var half_tile_size = tile_size / 2
 var level_num = 0
-var map = []
-var rooms = []
 var level_size
 var player_start_position
 
+var topLeftAnchorPoints = []
+var RoomCellSize = Vector2(9,9)
+var nextTileCellQueue = []
+var possibleStartPoints = []
+
+var Room_0000 = preload("res://Room_0000.tscn")
+var Corridor_1111 = preload("res://Corridor_1111.tscn")
+
+var CellMidpoints = {
+	"Top" : Vector2.ZERO,
+	"Bottom": Vector2.ZERO,
+	"Left": Vector2.ZERO,
+	"Right": Vector2.ZERO
+}
+
+var MapCornerPoints = {
+	"TopLeft": Vector2.ZERO,
+	"TopRight": Vector2.ZERO,
+	"BottomLeft": Vector2.ZERO,
+	"BottomRight": Vector2.ZERO	
+}
+
 const LEVEL_SIZES = [
-	Vector2(30, 30),
-	Vector2(35, 35),
-	Vector2(40, 40),
-	Vector2(45, 45),
-	Vector2(50, 50),
+	Vector2(54, 36),
+	Vector2(72, 54)
 ]
-const LEVEL_ROOM_COUNTS = [5, 7, 9, 12, 15]
-const MIN_ROOM_DIMENSION = 5
-const MAX_ROOM_DIMENSION = 8
 
 enum Tile {
-	Wall,	# 0
-	Floor,	# 1
-	SKIP,	# 2
-	Ladder,	# 3
-	Door,	# 4
-	Player,	# 5
-	Stone	# 6
+	BRCorner,	# 0
+	Door,	# 1
+	Floor,	# 2
+	VWall,	# 3
+	HWall,	# 4
+	TLCorner,	# 5
+	TRCorner, #6
+	TWall, #7
+	BLCorner, #8
+	Ladder
```

Comment:
I'd suggest using full words for describing enums.
Example:
`BRCorner`
would be:
`Bottom_Right_Corner`

Reply:
you mean BottomRightCorner?